src/basics.md

```diff
@@ -411,6 +412,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-simple-error-handling]: #ex-simple-error-handling
+<a name="ex-simple-error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Handles error that occur when we try to open a file that does not exist. We do this using [error-chain], a library that takes care of a lot of boilerplate code needed in order to [handle errors in Rust].
+
+We call `Io(std::io::Error)` inside `foreign_links` to convert the type `std::io::Error` into error-chain's custom version of it.
+
+To handle the result of a function that might return an error, in this example `open_foo`, it must return a `Result`. To handle that result, we use pattern matching on it when calling the function. In this example we print an error message if that result was an error.
+
+
+
+```rust
+#[macro_use]
+extern crate error_chain;
+
+error_chain!{
+    foreign_links {
+        Io(std::io::Error);
+    }
+}
+
+fn main() {
+    if let Err(ref e) = read_foo() {
+        // handle the error here
+        println!("error: {}", e);
```

Comment:
> This should use eprintln!()

Good point

> Also, I prefer the quick_main! macro(it also handles writing to stderr and panicking when it can't)

I would opt for explicit error handling this time for educational purposes as this aims to be a super basic example. Reader may refer to `quick_main!` usage in nearly every other example.

I would envision for the main here to mirror (but using `eprintln!`)
https://github.com/dtolnay/cargo-expand/blob/b05fcdabf48fa34c90343b1baa8777b9e65684db/src/main.rs#L18

which pretty much boils down to what `quick_main!` expands to.